src/compile/mark/mixins.ts

```diff
@@ -91,10 +92,36 @@ export function text(model: UnitModel, vgChannel: 'text' | 'tooltip' = 'text') {
 }
 
 export function bandPosition(channel: 'x'|'y', model: UnitModel) {
-  // TODO: band scale doesn't support size yet
   const fieldDef = model.encoding[channel];
   const scaleName = model.scaleName(channel);
   const sizeChannel = channel === 'x' ? 'width' : 'height';
+
+  if (model.encoding.size) {
+    const orient = model.markDef.orient;
+    if (orient) {
+      const centeredBandPositionMixins = {
+        // Use xc/yc and place the mark at the middle of the band
+        // This way we never have to deal with size's condition for x/y position.
+        [channel+'c']: ref.fieldRef(fieldDef, scaleName, {}, ref.band(scaleName, 0.5))
+      };
+
+      if (isFieldDef(model.encoding.size)) {
+        log.warn(log.message.cannotUseSizeFieldWithBandSize(channel));
+        // TODO: apply size to band and set scale range to some values between 0-1.
```

Comment:
I was looking into this but it requires some amount of refactoring in scale.  
Given this is a bad encoding anyway, I will focus on more important issues.